src/source.c

```diff
@@ -1858,13 +1858,27 @@ _dispatch_timers_get_delay(uint64_t nows[], struct dispatch_timer_s timer[],
 	return ridx;
 }
 
+
+#if HAVE_KEVENT64
```

Comment:
this block is wrong, and in any case should live in shims/time.h

I think what you mean here is "do we have support for NOTE_NSECONDS"
and not "do we have KEVENT64", you would break FreeBSD doing this.

Please do not mingle OS support with code, use shims/ and clean wrappers.

It'd be best to push NOTE_NSECOND support in libkqueue in the first place, which is probably easiest, so that you can use the thing directly anyway.